app/controllers/application_controller/ci_processing.rb

```diff
@@ -1609,42 +1607,55 @@ def scanclusters
     cluster_button_operation('scan', 'Analysis')
   end
 
+  def each_host(hosts, task_name)
+    Host.where(:id => hosts).order("lower(name)").each do |host|
+      begin
+        yield host
+      rescue StandardError => bang
+        add_flash(_("%{model} \"%{name}\": Error during '%{task}': %{message}") % {
+                    :model => ui_lookup(:model => "Host"), 
+                    :name => host.name, 
+                    :task => task_name,
+                    :message => bang.message 
+                  }, :error)
+      end
+    end
+  end
+
   # Common Host button handler routines
   def process_hosts(hosts, task, display_name = nil)
-    hosts, hosts_out_region = filter_ids_in_region(hosts, "Host")
+    hosts, _hosts_out_region = filter_ids_in_region(hosts, "Host")
     return if hosts.empty?
+    task_name = (display_name || task)
 
-    if task == "refresh_ems"
+    case task
+    when "refresh_ems"
       Host.refresh_ems(hosts)
-      #      add_flash("'" + task.titleize + "' initiated for " + pluralize(hosts.length,"Host"))
       add_flash(_("%{task} initiated for %{count_model} from the CFME Database") % {:task => (display_name || Dictionary.gettext(task, :type => :task).titleize), :count_model => pluralize(hosts.length, "Host")})
       AuditEvent.success(:userid => session[:userid], :event => "host_#{task}",
-          :message => "'#{display_name || task}' successfully initiated for #{pluralize(hosts.length, "Host")}",
+          :message => "'#{task_name}' successfully initiated for #{pluralize(hosts.length, "Host")}",
           :target_class => "Host")
-    elsif task == "destroy"
-      Host.where(:id => hosts).order("lower(name)").each do |host|
-        id = host.id
-        host_name = host.name
-        audit = {:event => "host_record_delete_initiated", :message => "[#{host_name}] Record delete initiated", :target_id => id, :target_class => "Host", :userid => session[:userid]}
+    when "destroy"
+      each_host(hosts, task_name) do |host|
+        audit = {:event => "host_record_delete_initiated", :message => "[#{host.name}] Record delete initiated", :target_id => id, :target_class => "Host", :userid => session[:userid]}
         AuditEvent.success(audit)
       end
       Host.destroy_queue(hosts)
+    when "scan"
+      each_host(hosts, task_name) do |host|
+        if host.respond_to?(:scan)
```

Comment:
@Fryguy : we have probably more candidates in the ci_processing that would be nice to have in the `is_available?` space.

Let's make an issue and get all of the done in a follow-up PR.